apps/OboeTester/app/src/main/cpp/InputStreamCallbackAnalyzer.cpp

```diff
@@ -22,35 +22,19 @@ oboe::DataCallbackResult InputStreamCallbackAnalyzer::onAudioReady(
         void *audioData,
         int numFrames) {
     int32_t channelCount = audioStream->getChannelCount();
-
     printScheduler();
-
-    if (audioStream->getFormat() == oboe::AudioFormat::I16) {
-        int16_t *shortData = (int16_t *) audioData;
-        if (mRecording != nullptr) {
-            mRecording->write(shortData, numFrames);
-        }
-        int16_t *frameData = shortData;
-        for (int iFrame = 0; iFrame < numFrames; iFrame++) {
-            for (int iChannel = 0; iChannel < channelCount; iChannel++) {
-                float sample = frameData[iChannel] / 32768.0f;
-                mPeakDetectors[iChannel].process(sample);
-            }
-            frameData += channelCount;
-        }
-    } else if (audioStream->getFormat() == oboe::AudioFormat::Float) {
-        float *floatData = (float *) audioData;
-        if (mRecording != nullptr) {
-            mRecording->write(floatData, numFrames);
-        }
-        float *frameData = floatData;
-        for (int iFrame = 0; iFrame < numFrames; iFrame++) {
-            for (int iChannel = 0; iChannel < channelCount; iChannel++) {
-                float sample = frameData[iChannel];
-                mPeakDetectors[iChannel].process(sample);
-            }
-            frameData += channelCount;
+    mInputConverter->convert(numFrames * channelCount, audioData);
+    float *floatData = (float *) mInputConverter->getOutputBuffer();
+    if (mRecording != nullptr) {
+        mRecording->write(floatData, numFrames);
+    }
+    float *frameData = floatData;
```

Comment:
It seems like `floatData` isn't modified anywhere else, so is there a need for `frameData`?